### PR TITLE
adding a sample Vagrantfile

### PR DESCRIPTION
**Pull request by fvanderbiest**
I think it may help, in addition with a README explaining how to proceed.

### CODE REVIEW

**Comment by landryb** [comment]
Will this ensure the ansible hosts file is correctly filled ?

**Comment by landryb** [comment]
Can you update it for jessie and test current master with it before i merge the PR ? It should deploy a fullblown instance from debian packages...

(I dont have virtualbox around, and i never tested vagrant ..)

**Comment by fvanderbiest** [comment]
Looks good, `vagrant up --provision` should do the trick.

**Comment by fvanderbiest** [comment]
Nope. I don't know yet if it's related with my PR, but the playbook fails at this step:
```
TASK: [postgresql | create georchestra user] ********************************** 
failed: [mygeorchestravm] => {"failed": true}
msg: unable to connect to database: could not connect to server: No such file or directory
	Is the server running locally and accepting
	connections on Unix domain socket "/var/run/postgresql/.s.PGSQL.5432"?


FATAL: all hosts have already failed -- aborting

PLAY RECAP ******************************************************************** 
           to retry, use: --limit @/home/fvanderbiest/georchestra.retry

mygeorchestravm            : ok=16   changed=14   unreachable=0    failed=1   

Ansible failed to complete successfully. Any error output should be
visible above. Please fix these errors and try again.
```

**Comment by landryb** [comment]
Jessie i suppose ? Is pgsql running ? Maybe there's a race between its installation/initialization and the user addition..

File: ansible.cfg [hunk 1]
```diff
@@ -1,4 +1,8 @@
 [defaults]
-hostfile=./hosts
+hostfile=./.vagrant/provisioners/ansible/inventory/vagrant_ansible_inventory
```
Comment (landryb) [inline ansible.cfg]: This will break the regular case :)

**Comment by landryb** [comment]
I suppose the hostname in the Vagrantfile (`"mygeorchestra"`) has to match the hostname in the inventory/hosts file ?

**Comment by fvanderbiest** [comment]
There were several issues:
 * first of all, the 5432 port was not forwarded
 * then pg had to be configured to listen to this port
 * finally the 9.4 cluster was no created due to an issue with locales (as reported [here](https://gist.github.com/jschoolcraft/1963369) for instance)

I guess the locale should be configurable in the playbook.

Still stuck later on with:
```
TASK: [postgresql | create missing schemas] *********************************** 
failed: [mygeorchestra] => (item={u'cmd': [u'psql', u'-d', u'georchestra', u'-t', u'-c', u"select * from pg_tables where schemaname='mapfishapp';"], u'end': u'2016-02-11 22:48:29.556310', u'stderr': u'', u'stdout': u'', u'changed': True, u'rc': 0, 'item': 'mapfishapp', u'warnings': [], u'delta': u'0:00:00.066637', 'invocation': {'module_name': u'command', 'module_args': u'psql -d georchestra -t -c "select * from pg_tables where schemaname=\'mapfishapp\';"'}, 'stdout_lines': [], u'start': u'2016-02-11 22:48:29.489673'}) => {"changed": true, "cmd": ["psql", "-d", "georchestra", "-f", "/tmp/tmp.cmO8LoYzDr/mapfishapp.sql"], "delta": "0:00:00.063775", "end": "2016-02-11 22:48:31.486482", "item": {"changed": true, "cmd": ["psql", "-d", "georchestra", "-t", "-c", "select * from pg_tables where schemaname='mapfishapp';"], "delta": "0:00:00.066637", "end": "2016-02-11 22:48:29.556310", "invocation": {"module_args": "psql -d georchestra -t -c \"select * from pg_tables where schemaname='mapfishapp';\"", "module_name": "command"}, "item": "mapfishapp", "rc": 0, "start": "2016-02-11 22:48:29.489673", "stderr": "", "stdout": "", "stdout_lines": [], "warnings": []}, "rc": 1, "start": "2016-02-11 22:48:31.422707", "warnings": []}
stderr: /tmp/tmp.cmO8LoYzDr/mapfishapp.sql: Permission denied
failed: [mygeorchestra] => (item={u'cmd': [u'psql', u'-d', u'georchestra', u'-t', u'-c', u"select * from pg_tables where schemaname='ldapadmin';"], u'end': u'2016-02-11 22:48:29.976678', u'stderr': u'', u'stdout': u'', u'changed': True, u'rc': 0, 'item': 'ldapadmin', u'warnings': [], u'delta': u'0:00:00.068345', 'invocation': {'module_name': u'command', 'module_args': u'psql -d georchestra -t -c "select * from pg_tables where schemaname=\'ldapadmin\';"'}, 'stdout_lines': [], u'start': u'2016-02-11 22:48:29.908333'}) => {"changed": true, "cmd": ["psql", "-d", "georchestra", "-f", "/tmp/tmp.cmO8LoYzDr/ldapadmin.sql"], "delta": "0:00:00.065739", "end": "2016-02-11 22:48:32.876041", "item": {"changed": true, "cmd": ["psql", "-d", "georchestra", "-t", "-c", "select * from pg_tables where schemaname='ldapadmin';"], "delta": "0:00:00.068345", "end": "2016-02-11 22:48:29.976678", "invocation": {"module_args": "psql -d georchestra -t -c \"select * from pg_tables where schemaname='ldapadmin';\"", "module_name": "command"}, "item": "ldapadmin", "rc": 0, "start": "2016-02-11 22:48:29.908333", "stderr": "", "stdout": "", "stdout_lines": [], "warnings": []}, "rc": 1, "start": "2016-02-11 22:48:32.810302", "warnings": []}
stderr: /tmp/tmp.cmO8LoYzDr/ldapadmin.sql: Permission denied
failed: [mygeorchestra] => (item={u'cmd': [u'psql', u'-d', u'georchestra', u'-t', u'-c', u"select * from pg_tables where schemaname='downloadform';"], u'end': u'2016-02-11 22:48:30.368118', u'stderr': u'', u'stdout': u'', u'changed': True, u'rc': 0, 'item': 'downloadform', u'warnings': [], u'delta': u'0:00:00.062597', 'invocation': {'module_name': u'command', 'module_args': u'psql -d georchestra -t -c "select * from pg_tables where schemaname=\'downloadform\';"'}, 'stdout_lines': [], u'start': u'2016-02-11 22:48:30.305521'}) => {"changed": true, "cmd": ["psql", "-d", "georchestra", "-f", "/tmp/tmp.cmO8LoYzDr/downloadform.sql"], "delta": "0:00:00.068053", "end": "2016-02-11 22:48:33.280207", "item": {"changed": true, "cmd": ["psql", "-d", "georchestra", "-t", "-c", "select * from pg_tables where schemaname='downloadform';"], "delta": "0:00:00.062597", "end": "2016-02-11 22:48:30.368118", "invocation": {"module_args": "psql -d georchestra -t -c \"select * from pg_tables where schemaname='downloadform';\"", "module_name": "command"}, "item": "downloadform", "rc": 0, "start": "2016-02-11 22:48:30.305521", "stderr": "", "stdout": "", "stdout_lines": [], "warnings": []}, "rc": 1, "start": "2016-02-11 22:48:33.212154", "warnings": []}
stderr: /tmp/tmp.cmO8LoYzDr/downloadform.sql: Permission denied
failed: [mygeorchestra] => (item={u'cmd': [u'psql', u'-d', u'georchestra', u'-t', u'-c', u"select * from pg_tables where schemaname='geofence';"], u'end': u'2016-02-11 22:48:30.730623', u'stderr': u'', u'stdout': u'', u'changed': True, u'rc': 0, 'item': 'geofence', u'warnings': [], u'delta': u'0:00:00.063677', 'invocation': {'module_name': u'command', 'module_args': u'psql -d georchestra -t -c "select * from pg_tables where schemaname=\'geofence\';"'}, 'stdout_lines': [], u'start': u'2016-02-11 22:48:30.666946'}) => {"changed": true, "cmd": ["psql", "-d", "georchestra", "-f", "/tmp/tmp.cmO8LoYzDr/geofence.sql"], "delta": "0:00:00.062979", "end": "2016-02-11 22:48:33.645516", "item": {"changed": true, "cmd": ["psql", "-d", "georchestra", "-t", "-c", "select * from pg_tables where schemaname='geofence';"], "delta": "0:00:00.063677", "end": "2016-02-11 22:48:30.730623", "invocation": {"module_args": "psql -d georchestra -t -c \"select * from pg_tables where schemaname='geofence';\"", "module_name": "command"}, "item": "geofence", "rc": 0, "start": "2016-02-11 22:48:30.666946", "stderr": "", "stdout": "", "stdout_lines": [], "warnings": []}, "rc": 1, "start": "2016-02-11 22:48:33.582537", "warnings": []}
stderr: /tmp/tmp.cmO8LoYzDr/geofence.sql: Permission denied
failed: [mygeorchestra] => (item={u'cmd': [u'psql', u'-d', u'georchestra', u'-t', u'-c', u"select * from pg_tables where schemaname='ogcstatistics';"], u'end': u'2016-02-11 22:48:31.103314', u'stderr': u'', u'stdout': u'', u'changed': True, u'rc': 0, 'item': 'ogcstatistics', u'warnings': [], u'delta': u'0:00:00.065979', 'invocation': {'module_name': u'command', 'module_args': u'psql -d georchestra -t -c "select * from pg_tables where schemaname=\'ogcstatistics\';"'}, 'stdout_lines': [], u'start': u'2016-02-11 22:48:31.037335'}) => {"changed": true, "cmd": ["psql", "-d", "georchestra", "-f", "/tmp/tmp.cmO8LoYzDr/ogcstatistics.sql"], "delta": "0:00:00.063470", "end": "2016-02-11 22:48:34.064510", "item": {"changed": true, "cmd": ["psql", "-d", "georchestra", "-t", "-c", "select * from pg_tables where schemaname='ogcstatistics';"], "delta": "0:00:00.065979", "end": "2016-02-11 22:48:31.103314", "invocation": {"module_args": "psql -d georchestra -t -c \"select * from pg_tables where schemaname='ogcstatistics';\"", "module_name": "command"}, "item": "ogcstatistics", "rc": 0, "start": "2016-02-11 22:48:31.037335", "stderr": "", "stdout": "", "stdout_lines": [], "warnings": []}, "rc": 1, "start": "2016-02-11 22:48:34.001040", "warnings": []}
stderr: /tmp/tmp.cmO8LoYzDr/ogcstatistics.sql: Permission denied

FATAL: all hosts have already failed -- aborting
```

**Comment by fvanderbiest** [comment]
> I suppose the hostname in the Vagrantfile ("mygeorchestra") has to match the hostname in the inventory/hosts file ?

Should be addressed with https://github.com/fvanderbiest/georchestra-ansible/commit/063b2426c06d8e2ccd60ab1bf7f2281b93375da8

**Comment by landryb** [comment]
i have to admit i didnt expect the pgsql port to be an issue. By default, it runs the db/user creation code (via psycopg2) on the remote host, and over the unix socket.. and by default, psql should listen on localhost:5432.

As for the locale thing, i'll add a task ensuring the locale is dpkg-reconfigure'd to en_US.UTF-8.

For the perms problem, i'll have a look at fetching the sql scripts as postgres user.

**Comment by landryb** [comment]
048a5bb should fix the locale issue.

**Comment by landryb** [comment]
Darn, hadnt seen that your PR had a better locale_gen fix for this... reverted.

**Comment by fvanderbiest** [comment]
> My point is - for users not using vagrant, i want it to work for them by default if they target a remote machine/lxc container. Vagrant-only settings belong to the Vagrantfile imo..

Right, should be fixed with aa679a8

**Comment by landryb** [comment]
Looks good :)

I only have to dive in the postgresql listen issue, there really should be no need to make it listen on more than localhost:5432. Maybe postgresql_db/postgresql_user actions can be done passing login_unix_socket=/var/run/postgresql/.s.PGSQL.5432 too..

Will try to test it with vbox this weekend.

**Comment by landryb** [comment]
Okay, struggled a bit with vagrant (had to install the package from the website, the 1.4 version shipped with my ubuntu LTS didnt work), but finally got to test it.

Commenting out the two postgresql-related tasks that you added, the db and users are properly created. By default postgresql listens on localhost, so that just works without fiddling with postgresql.conf/pg_hba.conf - no need for those tasks. Can you retest without them and the port 5432 forwarding ?

I'll fix the perms on the sql schemas, but other than that should be good to merge.

**Comment by landryb** [comment]
Merged, and removed the psql tasks in 5cba716. Fixed perms on sql schemas in 06f376a4 - now it should just work out of the box, even though i havent been able to fully test it to the end, it overflowed my disk with a 4Gb VM..

**Pull request merged**